src/java/org/apache/cassandra/audit/AuditLogOptions.java

```diff
@@ -37,11 +46,242 @@
+        private String archiveCommand;
+        private boolean block;
+        private long maxLogSize;
+
+        public Builder()
+        {
+            this(new AuditLogOptions());
+        }
+
+        public Builder(final AuditLogOptions opts)
+        {
+            this.enabled = opts.enabled;
+            this.logger = opts.logger;
+            this.includedKeyspaces = opts.included_keyspaces;
+            this.excludedKeyspaces = opts.excluded_keyspaces;
+            this.includedCategories = opts.included_categories;
+            this.excludedCategories = opts.excluded_categories;
+            this.includedUsers = opts.included_users;
+            this.excludedUsers = opts.excluded_users;
+            this.auditLogDir = opts.audit_logs_dir;
+            this.maxQueueWeight = opts.max_queue_weight;
+            this.maxArchiveRetries = opts.max_archive_retries;
+            this.rollCycle = opts.roll_cycle;
+            this.archiveCommand = opts.archive_command;
+            this.block = opts.block;
+            this.maxLogSize = opts.max_log_size;
+        }
+
+        public Builder withEnabled(boolean enabled)
+        {
+            this.enabled = enabled;
+            return this;
+        }
+
+        public Builder withLogger(final String loggerName, Map<String, String> parameters)
+        {
+
+            if (loggerName != null && !loggerName.trim().isEmpty())
+            {
+                this.logger = new ParameterizedClass(loggerName.trim(), parameters);
+            }
+
+            return this;
+        }
+
+        public Builder withIncludedKeyspaces(final String includedKeyspaces)
+        {
+            sanitise(includedKeyspaces).map(v -> this.includedKeyspaces = v);
+            return this;
+        }
+
+        public Builder withExcludedKeyspaces(final String excludedKeyspaces)
+        {
+            sanitise(excludedKeyspaces).map(v -> this.excludedKeyspaces = v);
+            return this;
+        }
+
+        public Builder withIncludedCategories(final String includedCategories)
+        {
+            sanitise(includedCategories).map(v -> this.includedCategories = v.toUpperCase());
+            return this;
+        }
+
+        public Builder withExcludedCategories(final String excludedCategories)
+        {
+            sanitise(excludedCategories).map(v -> this.excludedCategories = v.toUpperCase());
+            return this;
+        }
+
+        public Builder withIncludedUsers(final String includedUsers)
+        {
+            sanitise(includedUsers).map(v -> this.includedUsers = v);
+            return this;
+        }
+
+        public Builder withExcludedUsers(final String excludedUsers)
+        {
+            sanitise(excludedUsers).map(v -> this.excludedUsers = v);
+            return this;
+        }
+
+        public Builder withAuditLogDir(final String auditLogDir)
+        {
+            this.auditLogDir = auditLogDir;
+            return this;
+        }
```

Comment:
I just realized we don't use this one and I am wondering whether we expect to need It in the future. Not very strong opinion around that though. Maybe we can just suppress the warning for now:
```
**@SuppressWarnings("unused")**
public Builder withAuditLogDir(final String auditLogDir)
```
I leave it up to you.

Reply:
let's keep it how it is